test/functional/auth_source_ldaps_controller_test.rb

```diff
@@ -55,21 +55,27 @@ def test_destroy
     assert !AuthSourceLdap.exists?(auth_source_ldap.id)
   end
 
-  def setup_user
-    @request.session[:user] = users(:one).id
-    users(:one).roles       = [Role.default, Role.find_by_name('Viewer')]
-  end
+  context 'user with viewer rights' do
+    # The 'Viewer' role has '*_authorizers' filters, so these tests test
+    # the override of 'controller_name' to 'authorizers' instead of
+    # '*_auth_source_ldaps' permissions
+
+    setup do
+      @request.session[:user] = users(:one).id
+      users(:one).roles       = [Role.default, Role.find_by_name('Viewer')]
+    end
 
-  def user_with_viewer_rights_should_fail_to_an_edit_authentication_source
-    setup_user
-    get :edit, {:id => AuthSourceLdap.first.id}
-    assert @response.status == '403 Forbidden'
-  end
+    test 'should fail to edit authentication source' do
+      get :edit, { :id => AuthSourceLdap.first.id },
+        set_session_user(users(:one))
+      assert_response :forbidden
+      assert_includes @response.body, 'edit_authenticators'
+    end
 
-  def user_with_viewer_rights_should_succeed_in_viewing_authentication_sources
-    setup_user
-    get :index
-    assert_response :success
+    test 'should succeed in viewing authentication source' do
```

Comment:
This is quite useless test, index action returned 200 even before, the failure is only on granular filters but I know you didn't add it, so I leave it up to you if you want to remove it, improve it to test that user see some auth sources on page or just leave it as is.